### DISCUSSION TITLE
Lettore contatore GAS tramite protocollo IEC 62056 e ESPHOME

### DISCUSSION THREAD

**Discussion by array81** [indomus/forum#2387] [Hardware & Firmware]
Ho un contatore del gas della SAMGAS (modello RSE/2001LA) che dispone di una "porta infrarossi conforme allo standard EN 62056-21" (che poi credo sia in realtà IEC 62056-21).
Qui il [link ](https://www.fiorentini.com/wp-content/uploads/2021/03/rsen1_manual_ITA.pdf) del manuale.

Mi sto documentando sull'argomento per cercare di capire se sia fattibile realizzare un dispositivo (possibilmente controllato con ESPHOME o in alternativa con Tasmota) per misurare i consumi di GAS e integrarli con Home Assistant.
Da quanto ho capito il IEC 62056-21 è un protocollo universale usato per leggere i dati dei contatori (elettricità, corrente e gas) tramite una sonda esterna. Credo che uno possa anche auto costruirsi la sonda ma con una ventina di euro si trovano con tanto di anello metallico da agganciare al contatore (che presenta un magnete per tale scopo).
Cercando su internet o trovato un paio di "progetti" sull'argomento (anche se in quel caso erano contatori elettrici). La cosa non mi pare proprio semplicissima quindi prima di mettermi a studiare volevo vedere se qualcuno ci ha già provato.

Qualcuno ha esperienze in merito?

**Comment by aferrato** [comment]
Ciao, anche io ho quel contatore, ma ho letto nei manuali che la porta infrarossi è normalmente spenta e si attiva solo all'accensione del display. In più, se utilizzata invia comunicazione al gestore.

@array81 sei riuscito a leggere i consumi in altro modo?

**Reply by array81** [reply]
Si come detto la porta infrarossi si accende solo alla pressione di un tasto e si spegne dopo qualche secondo (non ricordo quanti ma pochi). Non penso sia fattibile a meno di installare un contatore tradizionale a valle e usare quello.
Personalmente ho rinunciato visto che ho consumi molto contenuti (uso il gas solo per cucinare), era infatti più un vezzo per avere anche quello oltre che acqua e elettricità su HA.

**Comment by Robyxxx** [comment]
Guardate qui: https://community.home-assistant.io/t/new-device-to-read-electric-meters-iec62056-21/624775

**Reply by array81** [reply]
L'idea è buona ma credo che il problema di fondo sia il contatore. Quello del post è un contatore di corrente elettrica che pertanto è "sempre acceso" cosi come probabilmente la porta a infrarossi. Nel caso del contatore di metano questo non è alimentato a corrente ma a batteria ragione per cui la porta infrarossi è normalmente spenta e si attiva solo alla pressione di un tasto per spegnersi dopo pochi secondi.
